apps/react-storybook/stories/chat/Chat.stories.tsx

```diff
@@ -214,37 +213,61 @@ export const DataLoading: Story = {
         height,
         disabled,
         rtlEnabled,
+        reloadOnChange,
         user,
         visible,
         hint,
         activeStateEnabled,
         hoverStateEnabled,
         focusStateEnabled,
-    }) => {       
-        const dataSource = new DataSource({
+    }) => {
+        const messagesRef = React.useRef([...initialMessages]);
+
+        const dataSource = useMemo(() => new DataSource({
             store: new CustomStore({
                 load: () => {
-                  const promise = new Promise((resolve) => {
-                    setTimeout(() => {
-                      resolve(initialMessages);
-                    }, 3000);
-                  });
-          
-                  return promise;
+                    const promise = new Promise((resolve) => {
+                        setTimeout(() => {
+                            resolve(messagesRef.current);
+                        }, 500);
+                    });
+            
+                    return promise;
+                },
+                insert: (message) => {
+                    if(reloadOnChange) {
```

Comment:
Why do we need it? I believe it should be executed in both cases

Reply:
When 'reloadOnChange' is true it requires to emulate data loading. I mean after call of built in reload() method we need to resolve() the load method with previous messages plus new one.
When 'reloadOnChange' is false the new message adds to the Chat after call store.push() method, and we render new message in list. It's not necessary to synchronize state with data here.

The main reason why I added this condition is to avoid double adding last item after switching the reloadOnChange mode using select panel in StoryBook after mutation of DataSource

Reply:
I gonna trust you about this